soh/src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -2224,11 +2239,27 @@ void EnXc_SetupDialogueAction(EnXc* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
         this->action = SHEIK_ACTION_IN_DIALOGUE;
     } else {
-        this->actor.flags |= ACTOR_FLAG_TARGETABLE | ACTOR_FLAG_FRIENDLY;
-        if (INV_CONTENT(ITEM_HOOKSHOT) != ITEM_NONE) {
-            this->actor.textId = 0x7010;
-        } else {
-            this->actor.textId = 0x700F;
+         this->actor.flags |= ACTOR_FLAG_TARGETABLE | ACTOR_FLAG_FRIENDLY;
+        if (gSaveContext.n64ddFlag && gPlayState->sceneNum == SCENE_TEMPLE_OF_TIME) {
+            if (!CHECK_DUNGEON_ITEM(DUNGEON_KEY_BOSS, SCENE_GANONS_TOWER)) {
+                this->actor.textId = 0x700F;    
```

Comment:
it'd be nice to use `#define`s or an enum for these instead of having them as magic numbers

Reply:
I agree, although this file had the original text IDs as magic numbers.